From c54da34236b10e0b172d7cb19e8f23f578ec8e8b Mon Sep 17 00:00:00 2001
From: Raveendra Babu Chennakesavulu <raveendra.babu.chennakesavulu@intel.com>
Date: Tue, 7 Jul 2020 23:27:37 +0530
Subject: [PATCH] Fix for adb failure in AIC

To configure ethernet proxy supplied from UI, ip link was made
false and true which is causing ip flush out and adb failure.

Fixed the issue by updating/configuring only proxy instead link
down/up

Change-Id: I8c1832436ea8d7c3438d9047950de68436308246
Tracked-On: OAM-91455
Signed-off-by: Raveendra Babu Chennakesavulu
                       <raveendra.babu.chennakesavulu@intel.com>

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index de0a691..0069cd1 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -269,6 +269,10 @@ public class EthernetNetworkFactory extends NetworkFactory {
         void setIpConfig(IpConfiguration ipConfig) {
 
             this.mIpConfig = ipConfig;
+            if (mIpConfig.getProxySettings() == ProxySettings.STATIC &&
+                    mIpClient != null) {
+                mIpClient.setHttpProxy(mIpConfig.getHttpProxy());
+            }
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
-- 
2.17.1

